### ISSUE TITLE
exporting filtered data

### ISSUE THREAD

**Issue by fpesciarelli** [jasp-stats/jasp-issues#461]
It could be useful if the exporting of data could take the filter into account. Maybe something like an "Export filtered" button.

For now, as suggested, I made the same formula in a computed column. Then I used the spreadsheet editor to remove all the "FALSE" lines in the filtered-column.
But, it will be much easier to export the data file with the filter applied.

Thanks!

**Comment by JorisGoosen** [comment]
A good idea!

**Comment by hanneoberman** [comment]
+1 on exporting the data with only the filtered rows (to then re-open JASP with just the filtered cases, which would significantly speed up all subsequent analyses with large $n$ datasets)

**Comment by tomtomme** [comment]
You can do this by clicking here: 

<img width="1915" height="1055" alt="Image" src="https://github.com/user-attachments/assets/795ffc90-45ae-4552-bccd-003c9b47ff88" />

This will remove all the values you do not want. Then save under a new file name.
Is this sufficient?

**Comment by hanneoberman** [comment]
Does this require clicking this button for each case (i.e., row/observation) or each category in the filter? 

In this specific use case, a student was working with large international panel data (>50.000 rows) and only needed a subset of national data (~5.000 rows).

**Comment by tomtomme** [comment]
for each category

**Comment by tomtomme** [comment]
So if I interpret your thumbs up correctly, this can be closed then. Please let me know if I am mistaken!

**Comment by hanneoberman** [comment]
Could you please explain to me how to save just the filtered rows, still? Clicking the 'remove' button only sets the category to missing, it seems.

<img width="899" height="1071" alt="Image" src="https://github.com/user-attachments/assets/be520834-700e-4170-8369-4445f9439e84" />

After removing category '3':

<img width="891" height="1051" alt="Image" src="https://github.com/user-attachments/assets/0a5776bc-4f3a-4b85-95de-c1cf1154185a" />

**Comment by tomtomme** [comment]
I probably still do not  undrstand what exactly you want. The 3 is gone. Just save the file now.
OR if you want to exclude and export the whole rows (cases), then do not use the X-button on the right, but the filter function in this way:

<img width="2461" height="1102" alt="Image" src="https://github.com/user-attachments/assets/055a6c7d-a120-407e-9f39-35f7fa235cc7" />

note that I have used the "eye" button above to "hide" the filtered rows. This way they are excluded on copy paste actions. You can of course also select all data including the header to get it copy pasted too

**Comment by hanneoberman** [comment]
Okay, thanks! The latter is indeed what this student needed. But I now understand that that is not possible with JASP only. Good to know there's a workaround via Excel!

**Comment by tomtomme** [comment]
No, you do not need Excel, only if you want to export and that is by definition stepping outside of jasp.
I do not understand probably, for what this would be needed. You can stay in JASP and analyze the filtered data as is, without copy pasting to Excel. For what does the student want to export said filtered data? Why does he need it outside of JASP?

**Comment by hanneoberman** [comment]
JASP kept crashing on their (quite old) machine with the large dataset. So, since they only needed to analyze a filtered subset of the data, we wanted to re-open JASP with just the filtered subset

**Comment by tomtomme** [comment]
got it. Valid use case. We also made improvements in recent JASP versions to handle big data - so you could also try out jasp 0.98

**Comment by hanneoberman** [comment]
Nice! Thank you for your advice and thinking along. We've got the workaround now, so no *need* to implement, and I'll check out the latest release 😄
